streams/src/main/java/org/apache/kafka/streams/tools/KafkaStreamsCleanupClient.java

```diff
@@ -0,0 +1,249 @@
+
+                    String retentionTime = topicRetentionTime.get(topic);
+                    if(retentionTime == null) {
+                        configs.remove("retention.ms");
+                    } else {
+                        configs.setProperty("retention.ms", retentionTime);
+                    }
+                    AdminUtils.changeTopicConfig(zkUtils, topic, configs);
+                } else {
+                    System.err.println("Intermediate user topic not found " + topic);
+                }
+            }
+
+            System.out.println("Done.");
+        } else {
+            System.out.println("No intermediate topics specified.");
+        }
+    }
+
+    private void deleteInternalTopics(List<String> allTopics) {
+        if (options.has(operator_option) || options.has(window_option)) {
+            System.out.println("Deleting internal topics.");
+
+            String applicationId = options.valueOf(application_id_option);
+
+            for (String operator : options.valuesOf(operator_option)) {
+                String changelog = applicationId + "-" + operator + "-changelog";
+                if(allTopics.contains(changelog)) {
+                    TopicCommand.main(new String[]{"--zookeeper", options.valueOf(zookeeper_option),
+                            "--delete", "--topic", changelog});
+                } else {
+                    System.err.println("No topic found for operator " + operator);
+                }
+            }
+
+            for (String window : options.valuesOf(window_option)) {
+                String changelog = applicationId + "-" + window + "-changelog";
+                if(allTopics.contains(changelog)) {
+                    TopicCommand.main(new String[]{"--zookeeper", options.valueOf(zookeeper_option),
+                            "--delete", "--topic", applicationId + "-" + window + "-changelog"});
+                } else {
+                    System.err.println("No topic found for window " + window);
+                }
+            }
+
+            System.out.println("Done.");
+        } else {
+            System.out.println("No operators or windows specified.");
+        }
+    }
+
+    private void removeLocalStateStore() {
+        File stateStore = new File(options.valueOf(state_dir_option) + File.separator + options.valueOf(application_id_option));
+        if(!stateStore.exists()) {
+            System.out.println("Nothing to clear. Local state store directory does not exist.");
+            return;
+        }
+        if(!stateStore.isDirectory()) {
+            System.err.println("ERROR: " + stateStore.getAbsolutePath() + " is not a directory.");
+            return;
+        }
+
+        System.out.println("Removing local state store.");
+        deleteDirectory(stateStore);
+        System.out.println("Deleted " + stateStore.getAbsolutePath());
+    }
+
+    private void deleteDirectory(File directory) {
```

Comment:
FWIW: Kafka has utility methods for removing dirs under `kafka.utils.CoreUtils.delete()` but IIRC that's part of the `core` dependency that we don't want to include...